brig/cmd/brig/commands/rerun.go

```diff
@@ -17,22 +20,39 @@ must still be accessible. Once a build is vacuumed, it can no longer be re-run.
 
 Using the '-f' flag will cause brig to resend the old payload, but override the old
 script with the provided one.
+
+As this command features all of the same flags as run, additional overrides are available,
+such as re-running with a different event, payload body, commit, etc.  See usage via -h/--help.
 `
 
 var (
-	rerunLogLevel string
-	rerunFile     string
+	rerunFile       string
+	rerunEvent      string
+	rerunPayload    string
+	rerunCommitish  string
+	rerunRef        string
+	rerunLogLevel   string
+	rerunNoProgress bool
+	rerunNoColor    bool
+	rerunBackground bool
 )
 
 func init() {
+	rerun.Flags().StringVarP(&rerunFile, "file", "f", "", "The JavaScript file to execute")
+	rerun.Flags().StringVarP(&rerunEvent, "event", "e", "", "The name of the event to fire")
+	rerun.Flags().StringVarP(&rerunPayload, "payload", "p", "", "The path to a payload file")
+	rerun.Flags().StringVarP(&rerunCommitish, "commit", "c", "", "A VCS (git) commit")
+	rerun.Flags().StringVarP(&rerunRef, "ref", "r", "", "A VCS (git) version, tag, or branch")
+	rerun.Flags().BoolVar(&rerunNoProgress, "no-progress", runNoProgress, "Disable progress meter")
+	rerun.Flags().BoolVar(&rerunNoColor, "no-color", runNoColor, "Remove color codes from log output")
+	rerun.Flags().BoolVarP(&rerunBackground, "background", "b", runBackground, "Trigger the event and exit. Let the job rerun in the background.")
```

Comment:
I think we'd better add a utility func that adds these flags to a given flagset, so that the func can be used from here and [run.go](https://github.com/brigadecore/brigade/blob/master/brig/cmd/brig/commands/run.go#L53) to reduce code duplication.

But that's an another story 😉

Reply:
Yes, good point.  Though there is a subtle difference in some of the flag's default values between rerun and run (for instance, we want the default rerun ref, commit, etc. to be empty so as not to override the initial run values), this perhaps could be hashed out in said utility function.  I agree we can save this for a later PR, ideally with retrofitting run.go with unit tests...